x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/Realms.java

```diff
@@ -368,8 +381,52 @@ public Map<String, Object> domainUsageStats() {
         }
     }
 
-    public Map<RealmConfig.RealmIdentifier, Authentication.RealmRef> getRealmRefs() {
-        return realmRefs;
+    /**
+     * Retrieves the {@link Authentication.RealmRef}, which contains the {@link DomainConfig}, if configured,
+     * for the passed in {@link RealmConfig.RealmIdentifier}.
+     * If the realm is not currently configured, {@code null} is returned.
+     */
+    public @Nullable Authentication.RealmRef getRealmRef(RealmConfig.RealmIdentifier realmIdentifier) {
```

Comment:
This is the goal of this PR.
It exposes the domain configuration (the set of realm ids) for a given realm id.
The plan is to use the realm id of the API Key owner to retrieve the domain using this method. Then  retrieve the profiles associated to any of the realms in the domain.